pkg/token/token.go

```diff
@@ -248,7 +248,11 @@ func (g generator) GetWithOptions(options *GetTokenOptions) (Token, error) {
 			profile = session.DefaultSharedConfigProfile
 		}
 		// create a cacheing Provider wrapper around the Credentials
-		if cacheProvider, err := filecache.NewFileCacheProvider(options.ClusterID, profile, options.AssumeRoleARN, sess.Config.Credentials); err == nil {
+		if cacheProvider, err := filecache.NewFileCacheProvider(
+			options.ClusterID,
+			profile,
+			options.AssumeRoleARN,
+			filecache.V1CredentialToV2Provider(sess.Config.Credentials)); err == nil {
```

Comment:
for my education: is the filecache intended to be v1 format until some future time?
reason for asking: at a future point(or in this PR) invalidate the filecache and use v2 format directly?

Reply:
Talked to Micah offline. The change as is is backward compatible. We can invalidate the cache if v1 creds are found in a later tag. The change will be backward incompatible because rollback will not work when filecache is sdkv2 creds.